src/Data/List/Linear.hs

```diff
@@ -204,6 +206,10 @@ intercalate sep = Unsafe.toLinear (NonLinear.intercalate sep)
 transpose :: [[a]] %1-> [[a]]
 transpose = Unsafe.toLinear NonLinear.transpose
 
+traverse' :: Data.Applicative f => (a %1-> f b) -> [a] %1-> f [b]
+traverse' _ [] = Data.pure []
+traverse' f (a:as) = (:) <$> f a <*> traverse' f as
```

Comment:
Just as a reference, this is relevant to https://github.com/tweag/linear-base/issues/190.